transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/EncodeTransformationPipelineBuilder.scala

```diff
@@ -103,4 +103,30 @@ object EncodeTransformationPipelineBuilder extends PipelineBuilder[Args] {
       submittedBy = donorInput.read[String]("submitted_by")
     )
   }
+
+  def transformAntibody(antibodyInput: Msg): Antibody = {
+    // Use regular expressions to remove everything but the actual target name, which should be the same across
+    // all targets in the list. Remove "synthetic_tag" type targets.
+    val mappedTarget = antibodyInput
+      .read[Array[String]]("targets")
+      .filterNot(target => target.matches(".*synthetic_tag/"))
+      .headOption
+      .map(target => target.replaceAll(raw"\/targets\/|-[a-zA-Z]*\/", ""))
```

Comment:
There's a slightly cleaner / more efficient way to do this.

First, you'll want to define a `val` outside of the method:
```scala
val EncodeIdPattern = "/[^/]+/([^/]+)/".r
```
Then this `map` can instead be a `collect`:
```suggestion
      .collect { case EncodeIdPattern(accession) => accession }
```

Reply:
I would also flip the `collect` and the `filter`, so the filter can then be a little simpler:
```scala
.filterNot(_.endsWith("synthetic_tag"))
```

Reply:
whoa fancy

Reply:
Oh I missed that this also needs to strip a suffix from the accession, hrm... I still like the idea of using the built-in regex matching capabilities, you'll just need to adjust the post-match logic. It can also double as the filter. Something like:
```
stuff
  .collect { moreStuff }
  .flatMap { accession =>
    val (front, back) = accession.split('-')
    if (back.endsWith(thing)) None
    else Some(front)
  }
```